### PR TITLE
Link ActivityType to Its Creator

### PR DESCRIPTION
**Pull request by amrro**
## Linear Ticket

[PRO-403 - Reporting Contribution doesn't link the ActivityTypes to its Creator/Guild](https://linear.app/govrn/issue/PRO-403/reporting-contribution-doesnt-link-the-activitytypes-to-its)

## Description

When creating a new `ActivityType`, it updates or link this to its creator by updating `UserActivity` Table. 

A new `getOrCreateActivityType` custom mutation has been added to handle creating `ActivityType`s.

### CODE REVIEW

**Comment by amsos** [comment]
quick question: with the pr does it update both the UserActivity and GuildActivityType table? 

I think when a user creates the activity, for now we just want it to link it to the user, not necessarily the guild.  

In previous versions, we've learned that auto updating it for the guild as well creates data explosions.  For example, 10 people all have a different way of writing `taking notes`.  And if we automatically update this as guild activity as well, the guild then has 10 different ways.

Consolidating and moving something from a user activity type to a guild activity type is actually something that the research team is working on.

**Comment by amrro** [comment]
Thanks a lot for your input. It answered a lot of my questions while implementing this. 
This is one of the reasons why I brought the idea of owner/admin/founder of a guild. 

I couldn't find a place that updates the Guild's `ActivityType`. Totally agree with you, this makes a lot of sense.

File: apps/protocol-api/src/prisma/resolvers/ActivityTypes.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        },
+        user: {
+          connect: {
+            id: args.data.userId,
+          },
+        },
+      },
+    });
+
+    return await prisma.activityType.findUnique({
```
Comment (alexkeating) [inline apps/protocol-api/src/prisma/resolvers/ActivityTypes.ts:67]: Does userActivity type not have an activity type field? We may be able to remove this request.

File: apps/protocol-api/src/prisma/resolvers/index.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
+import { ActivityTypesCustomResolver } from './ActivityTypes';
 import { AttestationResolver } from './Attestation';
 import { ContributionCustomResolver } from './Contribution';
 import { UserCustomResolver } from './User';
 
 export const customResolvers = [
+  ActivityTypesCustomResolver,
```
Comment (alexkeating) [inline apps/protocol-api/src/prisma/resolvers/index.ts:7]: We should rename the resolver to be singular. ActivityTypeCustomResolver

File: libs/protocol-client/src/lib/client/custom.ts [hunk 3]
```diff
@@ -16,6 +16,12 @@ export class Custom extends BaseClient {
   }
 
   public async createUserContribution(args: UserContributionCreateInput) {
+    await this.sdk.getOrCreateActivityType({
```
Comment (alexkeating) [inline libs/protocol-client/src/lib/client/custom.ts:19]: Should we get the response of this function call and update the args for createUserContribution?

**Review by alexkeating — COMMENTED** [review]
Some small comments

**Pull request merged**